### PR TITLE
Write initial ERC1319URI class

### PR DESCRIPTION
**Pull request by njgheorghita**
Class for handling erc1319 URIs was needed. 

example Registry URI:
`erc1319://address:chainId/packageName@version`

### CODE REVIEW

File: src/utils/uri.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+// i.e. 
+// erc1319://snakecharmers.eth:1/dai@1.0.0
+export class Erc1319URI {
+  scheme: string;
+  address: string;
+  chainId: number;
+  packageName: string;
+  version: string;
+
+  constructor (uri: any) {
```
Comment (gnidan) [inline src/utils/uri.ts]: Can `any` just be `string`?

File: src/utils/uri.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  if(chainId != 1){
+	throw new Error(
+	  `Invalid chain ID: ${chainId}. Currently only mainnet is supported.`
+	)
+  } else {
+	return chainId
+  }
+}
+
+function parsePackageId(url: URL) {
```
Comment (gnidan) [inline src/utils/uri.ts]: these functions can probably just be static methods on the class https://www.typescriptlang.org/docs/handbook/classes.html#static-properties

File: src/utils/uri.ts (hunk 1 above)
Comment (gnidan) [reply src/utils/uri.ts]: Alternatively: it could be `string | URL`, and then check with `typeof uri === "string"` to see if you need to do `new URL`

**Comment by njgheorghita** [comment]
Dope! thanks for the tips @gnidan

**Pull request merged**